proj/src/main/webapp/script.js

```diff
@@ -27,17 +29,101 @@ if(btnLogin){
     });
 }
 
+
+// Adds user to an existing chat when given a reference to the place in the database
+function addUserToTag(reference){
+    var currentUID = firebase.auth().currentUser.uid
+    console.log("adding new user to chat room with uid: " + currentUID);
+    reference.push(currentUID);
+}
+
+// Creates a new chat given a tag and adds the current user as a member
+function createNewChatWithUser(tag){
+    console.log("creating new chat with tag: " + tag);
+    var newChat = {
+        "name" : tag,
+        "tag" : tag,
+    };
+    var currentReference = firebase.database().ref("/chat/" + tag);
+    var postKey = currentReference.push(newChat).key;
+    currentReference = firebase.database().ref("/chat/" + tag + "/" + postKey + "/users/");
```

Comment:
Didn't arrington make a helper method for concatenating the currentTag and returning the reference or something close? I think it might be good to use that here so that if the path to chats changes (e.g. `/chat/tag/key/users/` -> `/messages/tag/key/users/` you will only need to change it in one place.